### PR TITLE
Fix memoization problems

### PR DESCRIPTION
**Pull request by AbanoubGhadban**
Fixes some memoization problems:
- It memoize the function that lazily loads the module, so it doesn't call it many times.
- Recall the function when the cache key is changed

### CODE REVIEW

File: packages/macro/src/index.macro.ts [hunk 1]
```diff
@@ -12,7 +12,18 @@ function addImportStatement(importName: string, importPath: string, isDefault: b
...
 
   const existingImportDeclaration = programPath.node.body.find(
-    p => t.isImportDeclaration(p) && p.source.value === importPath
+    p => {
+      // return t.isImportDeclaration(p) && p.source.value === importPath;
+      if (!t.isImportDeclaration(p) || p.source.value !== importPath) {
+        return false;
+      }
+
+      if (isDefault) {
```
Comment (AbanoubGhadban) [inline packages/macro/src/index.macro.ts:21]: It ensures that the import statement exists and imports the exact element I want to import. **It checks both the import path and the import name**. It fixes the problem when I try to make the plugin import `useMemo` when react is already imported.

File: packages/macro/tests/__snapshots__/index.test.ts.snap [hunk 2]
```diff
@@ -81,16 +92,20 @@ exports[`macros 3. custom-wepback-chunk-client-side: 3. custom-wepback-chunk-cli
...
-    import(
-      /* webpackChunkName: "custom-chunk-name-ssr-computations" */ "./a.ssr-computation"
-    ),
+  useCallback(
+    () =>
+      import(
+        /* webpackChunkName: "custom-chunk-name-ssr-computations" */ "./a.ssr-computation"
+      ),
+    [],
+  ),
```
Comment (Romex91) [inline packages/macro/tests/__snapshots__/index.test.ts.snap:108]: Can we define this function at the module scope and used it without `useCallback`?
E.G:
```
import { useCallback } from "react";
import useSSRComputation_Client from "use-ssr-computation.runtime/lib/useSSRComputation_Client";
function importCustomChunkNameSsrComputation() {
  return import(/* webpackChunkName: "custom-chunk-name-ssr-computations" */ "./a.ssr-computation");
}

// ...

const x = useSSRComputation_Client(importCustomChunkNameSsrComputation,[]),
```

**Review by Romex91 — COMMENTED** [review]
Looks incredible. Single suggestion.

File: packages/macro/tests/__snapshots__/index.test.ts.snap (hunk 2 above)
Comment (AbanoubGhadban) [reply packages/macro/tests/__snapshots__/index.test.ts.snap]: @Romex91 Done

**Review by Romex91 — APPROVED** [review]
Great!

**Pull request merged**